### PR TITLE
Fix flaky test by using assertBusy to avoid NPE

### PR DESCRIPTION
**Pull request by matriv**
@mfussenegger plz review

### CODE REVIEW

File: sql/src/test/java/io/crate/operation/collect/ShardCollectServiceTest.java [hunk 1]
```diff
@@ -45,41 +46,54 @@
...
         upload("b1", "foo");
         upload("b1", "bar");
+        ensureGreen();
+
+        assertBusy(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    IndexService indexService = internalCluster().getInstance(IndicesService.class).indexService(".blob_b1");
+                    assertNotNull(indexService);
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/operation/collect/ShardCollectServiceTest.java]: does really everything have to be insert the assertBusy?

Comment (matriv) [inline sql/src/test/java/io/crate/operation/collect/ShardCollectServiceTest.java]: I wanted to make sure that I get the shardCollectService.
So I would need to define the vars (indexService, etc) before assert busy and then 
create a private class with constructor that updates them and pass it as runnable).
I prefered to use the anonymous class but then everything must be inside.

Any better suggestion?

Comment (matriv) [inline sql/src/test/java/io/crate/operation/collect/ShardCollectServiceTest.java]: btw: this test never failed locally with 10000 iterations,
and by running the ./gradlew -PtestForks=2 test at least 5 times.

Comment (matriv) [inline sql/src/test/java/io/crate/operation/collect/ShardCollectServiceTest.java]: Alternative? http://pastie.org/private/ibulviqbubzirli42texdg

**Comment by mfussenegger** [comment]
lgtm

**Pull request merged**